### PR TITLE
Edit start script for checking environment once.

### PR DESCRIPTION
**Pull request by liaosunny123**
### Background
#2901 , reducing the start time.  

### Changes
Modifying the `run.bat` and `run.sh`. I add the command to generate a file named `env_check.lock` to skip the env check after first starting.  

### Documentation
As what the changes said.  

### Test Plan
I tested in window and ubuntu 22.04, it works.  

### PR Quality Checklist
- [√] My pull request is atomic and focuses on a single change.
- [√] I have thoroughly tested my changes with multiple different prompts.
- [√] I have considered potential risks and mitigations for my changes.
- [√] I have documented my changes clearly and comprehensively.
- [√] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,9 +1,12 @@
 #!/bin/bash
-python scripts/check_requirements.py requirements.txt
-if [ $? -eq 1 ]
-then
+if [ ! -d "./env_check.lock" ];then
```
Comment (AndresCdo) [inline run.sh]: Use variable to define file name.

Comment (AndresCdo) [inline run.sh]: Change the check for the lock file from a directory (-d) to a regular file (-f).
```suggestion
if [ ! -f "./env_check.lock" ];then
```

File: run.sh [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+    python scripts/check_requirements.py requirements.txt
+    if [ $? -eq 1 ]
+    then
     echo Installing missing packages...
     pip install -r requirements.txt
+    fi
+    touch env_check.lock
 fi
 python -m autogpt $@
```
Comment (AndresCdo) [inline run.sh]: Replaced the variable _$@_ with _"$@"_ to properly handle arguments containing spaces.
```suggestion
python -m autogpt "$@"
```

File: run.sh (hunk 1 above)
Comment (liaosunny123) [reply run.sh]: ok, i add it.

**Comment by hdkiller** [comment]
The script is intended to run the check requirements every time in case the repository was pulled by the user and there are new requirements added, so a one time lock file is not enough.

A better solution would be to check if the requirements file has been not changed since a successful pip install.

**Comment by liaosunny123** [comment]
> The script is intended to run the check requirements every time in case the repository was pulled by the user and there are new requirements added, so a one time lock file is not enough.
> 
> A better solution would be to check if the requirements file has been not changed since a successful pip install.

You are right and I ignore this situation. I check `check_requirements.py` and I think it is unnecessary to get the missing package exactly. Do you think check the hash of the requirment file to decide whether install requirement is a good idea?

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by ntindle** [comment]
A hash would work well in this case I believe as we typically pin versions of something breaks relating to a package

**Comment by liaosunny123** [comment]
check `requirement.py` hash to decide if need to procee

**Pull request closed without merging**